### ISSUE TITLE
Any plans to make new release?

### ISSUE THREAD

**Issue by kloczek** [tesseract-ocr/tesseract#3331]
I think that it would be good to flush currently committed changes and make new release :)

**Comment by GuillemGSubies** [comment]
I would love to see a new release too

**Comment by Shreeshrii** [comment]
Please see https://tesseract-ocr.github.io/tessdoc/Planning for some earlier discussion.

**Comment by amitdo** [comment]
@stweil,

What about tagging a new alpha soon?

**Comment by stweil** [comment]
That's my plan. I just want to make sure that it does not contain major issues.

**Comment by stweil** [comment]
See [new release](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-alpha-20210401).

**Comment by kloczek** [comment]
- looks like it not reccomended to use as stable release. Am I right?
- could you please concuder change tagging convention and not use dashesh in version string?
None of the packaging software allows use dashes in version string (simple iusing not standard versions strings adds kind of obstacle on automated packaging). With using only digits and dots exampl version string for alpha would be like `4,99.1` this would allow as well easy upgrade when `5.0.0` finally will be released :)

**Comment by stweil** [comment]
I recommend using the latest version of Tesseract because it is faster and has less errors than release 4.1.1. It is nevertheless tagged as alpha prerelease because there are still changes planned for API and functionality. Those changes don't affect most normal users of Tesseract.

The version string follows the rules for semantic versioning and should be easy to understand and parse. Ubuntu has much more complex package versions like `0.5.3+18.04.20180207.2-0ubuntu2` for example.

**Comment by kloczek** [comment]
Please be aware that tesseract is not only packaged by Ubuntu and not as well only Linux
If just released version is recommended as stable why not just vrersion it as `5.0.0`?

**Comment by stweil** [comment]
>  If just released version is recommended as stable why not just version it as `5.0.0`?

As soon as there is a 5.0.0, any more API changes would require to switch to 6.0.0, 7.0.0 and so on.

**Comment by amitdo** [comment]
Last release was in December 2019. And it was a partial backport from 5.0.0.

In many workplaces, it is forbidden to use a version of software which is in "alpha" or "beta" state.

Even outside of work environment, many users will stay away from "alpha" version.

Many of our users are waiting for a new, improved release, which is not tagged as alpha/beta.

I suggest to release a new "final" 5.0.0 version in the next 30-90 days.

My 2 cents...

**Comment by stweil** [comment]
> I suggest to release a new "final" 5.0.0 version in the next 60-120 days.

So let's see how far we come with the to do issues from the [Tesseract next project list](https://github.com/tesseract-ocr/tesseract/projects/1).

**Comment by egorpugin** [comment]
Some of them are blocked by 5.0.0, so rational separation between 5 and 6 should be done.
My vote is to keep 5.0 small (current state is ok) and unblock 6.0 faster and 6.0 will have more changes.

**Comment by amitdo** [comment]
Can we move to 'beta' before the of this month?

**Comment by amitdo** [comment]
@stweil, when can we release 5.0 final? First step should be 5.0.0-beta.

**Comment by stweil** [comment]
I just tagged [5.0.0-beta-20210815](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-beta-20210815).

**Comment by nagadomi** [comment]
Is there any chance that a FAST_FLOAT enabled Windows binary will be provided in the future? (or FAST_FLOAT will be the default)
Should I decide to build it myself?

**Comment by stweil** [comment]
Our latest installers for Windows use FAST_FLOAT, see https://github.com/UB-Mannheim/tesseract/wiki#history.

**Comment by nagadomi** [comment]
I had missed that comment. Thank you.

**Comment by stweil** [comment]
Our [milestone for 5.0.0](https://github.com/tesseract-ocr/tesseract/milestone/6) still includes 21 open issues.

We have to decide which of those issues should be fixed for the final 5.0.0 release and which issues can be fixed later, either in a 5.x.y or in a future release 6.

Please add your opinions at the different issue discussions.

**Comment by zdenop** [comment]
We are at the stage, that I suggest to use/test current master in case of any problem with version 4. Backport fixes and improvement from version 5 is not worthy of time and energy IMO.   I am convinced we should release version 5 ASAP (e.g. in end of September to have change make inevitable fixes, tests and docs adjustments).

I would suggest to use following prioritization of issues:
1. Easy fixes ;-) (e.g. if somebody is working on something)
2. API changes (e.g. missing table extraction is really big topic)
3. General problem (e.g. changes related to only particular language /OS could be postponed)

**Comment by amitdo** [comment]
@zdenop or @stweil, could you announce the beta pre-release in the tesseract-ocr forum? Ask users to test it (or the master branch) and report about any regression in speed or accuracy compared to **4.1.1**.

**Comment by amitdo** [comment]
https://groups.google.com/g/tesseract-ocr/c/pd_8B0wGBZc

Zdenko, thanks!

**Comment by amitdo** [comment]
https://www.phoronix.com/scan.php?page=news_item&px=Tesseract-5.0-Beta

**Comment by stweil** [comment]
https://twitter.com/stjoweil/status/1426142063167516672

**Comment by stweil** [comment]
Should we also make a new release 4.1.2, based on the current [4.1 branch](https://github.com/tesseract-ocr/tesseract/tree/4.1) or with additional backports (which ones)?

**Comment by amitdo** [comment]
I suggest to focus on 5.0.0 only.

**Comment by egorpugin** [comment]
Also I'd like to focus on future open-to-changes-master-branch and modernization efforts.

**Comment by amitdo** [comment]
Can we move to rc1 in this week?

**Comment by stweil** [comment]
I hope so, but there are still some open issue with crashes.

**Comment by stweil** [comment]
I plan to tag the next release on 2021-09-15 (one month after the last one). Should it be `5.0.0-beta-20210915`, or should it be `5.0.0-rc1`?

**Comment by amitdo** [comment]
I vote for beta2. 2-3 weeks later rc1. 2-3 weeks later 5.0.0 (or rc2).

**Comment by kloczek** [comment]
Please do not use dashes in tags because this makes versionng harder on packagiun layer as none of the package mangement software accepts dahes in version strins.
Please use `5.0.beta`, `5.0.rc1` and finaly `5.0.0`.
This will make tagginng packaging fiendly because digits are at the end of the alphabet and this wouls allow upgrades `5.0.beta` -> `5.0.rc1` -> `5.0.0`.
Gnome uses now that pattern and it would be good if more maintainers would be using the same tagging convention  https://discourse.gnome.org/t/new-gnome-versioning-scheme/4235

**Comment by egorpugin** [comment]
But tess uses semver, as I understand.

**Comment by stweil** [comment]
Yes, that's right. And our tag names also follow the rules of semver.

**Comment by kloczek** [comment]
Are you sure? :P
```console
[tkloczko@barrel tesseract]$ git tag |grep -- -
3.03-rc1
3.04-rc1
4.0.0-alpha
4.0.0-beta.1
4.0.0-beta.2
4.0.0-beta.3
4.0.0-beta.4
4.0.0-rc1
4.0.0-rc2
4.0.0-rc3
4.0.0-rc4
4.1.0-rc1
4.1.0-rc2
4.1.0-rc3
4.1.0-rc4
4.1.1-rc1
4.1.1-rc2
5.0.0-alpha
5.0.0-alpha-20201224
5.0.0-alpha-20201231
5.0.0-alpha-20210401
5.0.0-beta-20210815
```

**Comment by bertsky** [comment]
BTW considering that this went undetected ever since 4.0 came to be, and custom training is essential for very high quality, I really think that https://github.com/tesseract-ocr/tesstrain/issues/261 is urgent. (It [really belongs here](https://github.com/tesseract-ocr/tesstrain/issues/261#issuecomment-880631305), not to tesstrain.) It should be addressed in the final 5.0 in some way (even if it's just a warning that lstmtraining's checkpoint selection should not be used and all of its CER values should be ignored for now).

**Comment by egorpugin** [comment]
> Are you sure? :P

Yes, since some 4 release.

**Comment by stweil** [comment]
@kloczek, I am not sure what you want to say. Using dashes is permitted by semver. And as Egor said, Tesseract used a different convention until `4.00.00alpha` and `4.00.00dev`, but then switched to semver with `4.0.0-alpha`.

**Comment by kloczek** [comment]
I'm talking not about servers (whatever it means) but packaging software like rpm., dpkg, Solaris IPS etc.

**Comment by stweil** [comment]
Such packaging software normally uses stable releases like 4.1.1 or the planned 5.0.0.

If packaging software uses alpha or beta versions, it should be prepared to do so. Here is an example for ArchLinux which normally does that: https://wiki.archlinux.org/title/VCS_package_guidelines#Git.

**Comment by kloczek** [comment]
You are talking about what typically happens on OS dostribution layer.
Distribution consist of packages. Sometiimes versions of the software marked as stable/release are unstable :P

I'm talking aboutyusing version schema wchi allows upgrade from unstable to stable without uninstalling or fotcing to ignore lower version string.

**Comment by stweil** [comment]
There is now a new release [5.0.0-beta-20210916](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-beta-20210916).

A [recent build issue](https://github.com/tesseract-ocr/tesseract/issues/3527#issuecomment-919987660) and some other bugs should be fixed before we can create a release candidate for the final 5.0.0 release.

**Comment by kloczek** [comment]
Still tagging not firendly for packaging :/

**Comment by stweil** [comment]
Wait for release `5.0.0`. We follow the semver standard, and packaging which supports that standard should not have problems. Regarding tag names, there are many different conventions. Some use the version also for the tag, others add a leading character like `v` or `V`, some use underscores, minus or points. Up to now I did not find a generally accepted rule how to get the git tag from a version.

**Comment by stweil** [comment]
I'd like to tag a new release 4.1.2 based on the [4.1 branch](https://github.com/tesseract-ocr/tesseract/tree/4.1). See the [list of commits](https://github.com/tesseract-ocr/tesseract/compare/4.1.1...4.1) for all changes since the last stable [release 4.1.1](https://github.com/tesseract-ocr/tesseract/releases/tag/4.1.1).

Most Linux distributions still provide 4.1.1, and some (for example Debian) won't update to the planned 5.0.0 soon. So for those distributions bug fix releases like a new 4.1.2 are important.

@AlexanderP, would you like to have any other changes (for example to require less patches for making the Debian package)?

**Comment by zdenop** [comment]
@stweil : there were several reports for 4.x for build system (cmake and  autotols visibility) that seem to be solved in 5.x.  This should be backported to 4.1.2 IMO.
I hope I can fix cmake and cygwin this weekend #2379 + add libcurl support to cmake.

**Comment by stweil** [comment]
Then I wait for your backports before tagging 4.1.2.

**Comment by AlexanderP** [comment]
@stweil not at the moment

**Comment by amitdo** [comment]
When are we moving to `rc1`?

With rc1, we should do only bug fixes.

**Comment by stweil** [comment]
Release [5.0.0-rc1](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-rc1) is now available.

**Comment by stweil** [comment]
I'd like to release 5.0.5-rc2 this weekend. Is there anything open which should be done before that release?

**Comment by Shreeshrii** [comment]
@stweil If possible, I have a small feature request.

Currently, newly trained traineddata files display version string which reflects the tesseract version they were trained with. Is it possible to also include some other identifying features which are available in the network spec display also in version string?

In the past, I have added custom strings to it, but it would be nice to have it done bt default.

See https://github.com/Shreeshrii/tesstrain-Sanskrit-IAST/blob/main/README.md for some examples. The versions in finetuned traineddata from different checkpoints of same training is same. Would be good if there can be some differentiation.

**Comment by amitdo** [comment]
I opened #3637 for  @Shreeshrii's feature request.

**Comment by stweil** [comment]
Release [5.0.0-rc2](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-rc2) is now available.

**Comment by stweil** [comment]
> 4.1.2



> Should we also make a new release 4.1.2, based on the current [4.1 branch](https://github.com/tesseract-ocr/tesseract/tree/4.1) or with additional backports (which ones)?

Release [4.1.2](https://github.com/tesseract-ocr/tesseract/releases/tag/4.1.2) is now available.

**Comment by stweil** [comment]
I am sorry, but the 4.1.2 is broken (issue #3642). That was fixed in commit 28ad3e8b990e930920d9dac3efca09223a8c3b16. So we need a new release 4.1.3. Is there anything else which should be included in 4.1.3?

**Comment by stweil** [comment]
Release [4.1.3](https://github.com/tesseract-ocr/tesseract/releases/tag/4.1.3) is now available.

**Comment by kloczek** [comment]
LGTM :)
Thank you.

Closing.

**Comment by amitdo** [comment]
@stweil, what's the next step? Another RC or final 5.0.0 release? When?

**Comment by stweil** [comment]
Do you think that the release is good enough for a final 5.0.0? What would you suggest?

I just recently noticed a new bug #3655. And I still wait for reviews of pull request #3651 which is an API change, so must be merged in 5.0.0. Therefore a 5.0.0-rc3 might be reasonable.

**Comment by amitdo** [comment]
>Do you think that the release is good enough for a final 5.0.0?

Yes.

>I just recently noticed a new bug #3655.

Do you think you can fix it this week?

 >I still wait for reviews of pull request #3651 which is an API change, so must be merged in 5.0.0.

If you don't get reply in the next few days, I suggest to merge this PR.

>Therefore a 5.0.0-rc3 might be reasonable.

Okay, but let's release 5.0.0 this year...

**Comment by egorpugin** [comment]
You can post-fix any wanted issues into minor releases (5.1 or 5.0.1).

**Comment by stweil** [comment]
Release [5.0.0-rc3](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-rc3) is now available.

**Comment by amitdo** [comment]
@egorpugin,

IMO, we should test 5.0.0 with VS2022 and still test VS2019.

https://devblogs.microsoft.com/visualstudio/visual-studio-2022-now-available/

https://github.blog/changelog/2021-08-23-github-actions-windows-server-2022-with-visual-studio-2022-is-now-available-on-github-hosted-runners-public-beta/

https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md

**Comment by egorpugin** [comment]
See https://github.com/tesseract-ocr/tesseract/commit/6f399c0df1bc4f4337a85babb0e56b49f2f56777

**Comment by stweil** [comment]
I'd like to release 5.0.0 today (within the next few hours). Is there anything still missing for the release?

There are still several open issues, but those can be addressed in 5.0.1 or 5.1.0.

**Comment by Shreeshrii** [comment]
Thank you @stweil and other contributors for 5.0.0.

**Comment by stweil** [comment]
There where nearly 70 contributors from 4.0 to 5.0.0, and about 30 of them had at least two contributions. That's nice.

**Comment by stweil** [comment]
[Release 5.0.0](https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0) is now available.

**Comment by VV-KAR** [comment]
Congrats on the release! Do you have an eta on the exe for 5.0.0? Thanks!

**Comment by stweil** [comment]
Do you mean https://github.com/UB-Mannheim/tesseract/wiki?

**Comment by amitdo** [comment]
5.x (post-5.0.0) and 6.x plans should be discussed in separates issues.

**Comment by stweil** [comment]
https://twitter.com/stjoweil/status/1465749015656644608
https://www.phoronix.com/scan.php?page=news_item&px=Tesseract-5.0-Released

**Comment by Shreeshrii** [comment]
The readme, documentation and tesstrain should also be updated for 5.0.0. Afterwards tessdoc and tesstrain should also be given a 5.0 version.

**Comment by stweil** [comment]
tessdata* also need new tags, but first we should review and fix some open issues.

**Comment by Shreeshrii** [comment]
Please see https://groups.google.com/g/tesseract-ocr/c/7nvw7qG8lWc/m/LnmSvFKDBQAJ

```
marco atzeri
Dec 4, 2021, 10:37:33 PM (8 hours ago) 
to tesser...@googlegroups.com
Hi,

I am updating the cygwin package from 4.1.1 to 5.0.0
and I noticed that 3 scripts

language-specific.sh
tesstrain.sh
tesstrain_utils.sh

are not anymore in the source code.

Have they been replaced by something else that I should pack instead ?

Regards
Marco
```

**Comment by amitdo** [comment]
> Release [4.1.3](https://github.com/tesseract-ocr/tesseract/releases/tag/4.1.3) is now available.

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8-beta/html-single/8.5_release_notes/index

Let's see if they will upgrade to 4.1.3 in RHEL 8.6, and if they will ship Tesseract 5.0.0 in RHEL 9.

**Comment by amitdo** [comment]
@stweil, can we close this issue?

I think 4.1.3 should be the last version in the 4.x series.
